In [4]:
import joblib
import pandas as pd

# Load the trained model and preprocessing objects
model = joblib.load('model.pkl')
scaler = joblib.load('scaler.pkl')
label_encoder = joblib.load('label_encoder.pkl')

# Define a function for preprocessing the input data
def preprocess_input(data):
    # Apply the same preprocessing steps as before
    scaled_data = scaler.transform(data)
    return scaled_data

# Define a function for making predictions
def predict_species(data):
    # Preprocess the input data
    preprocessed_data = preprocess_input(data)
    
    # Make predictions using the loaded model
    predictions = model.predict(preprocessed_data)
    
    # Inverse transform the predicted labels back to original class names
    predicted_species = label_encoder.inverse_transform(predictions)
    
    return predicted_species

# Get input values from the user
sepal_length = float(input("Enter sepal length: "))
sepal_width = float(input("Enter sepal width: "))
petal_length = float(input("Enter petal length: "))
petal_width = float(input("Enter petal width: "))

# Create a DataFrame with user input
input_data = pd.DataFrame([[sepal_length, sepal_width, petal_length, petal_width]], columns=['sepal_length', 'sepal_width', 'petal_length', 'petal_width'])

# Call the predict_species function with user input data
predictions = predict_species(input_data)
print("Predicted species:", predictions[0])


Enter sepal length: 2.2
Enter sepal width: 2.2
Enter petal length: 2.2
Enter petal width: 2.2
Predicted species: versicolor


In [5]:
import joblib
import pandas as pd

# Load the trained model and preprocessing objects
model = joblib.load('regression_model.pkl')
scaler = joblib.load('scaler.pkl')

# Define a function for preprocessing the input data
def preprocess_input(data):
    # Apply the same preprocessing steps as before
    scaled_data = scaler.transform(data)
    return scaled_data

# Define a function for making predictions
def predict_species(data):
    # Preprocess the input data
    preprocessed_data = preprocess_input(data)
    
    # Make predictions using the loaded model
    predictions = model.predict(preprocessed_data)
    
    return predictions

# Get input values from the user
sepal_length = float(input("Enter sepal length: "))
sepal_width = float(input("Enter sepal width: "))
petal_length = float(input("Enter petal length: "))
petal_width = float(input("Enter petal width: "))

# Create a DataFrame with user input
input_data = pd.DataFrame([[sepal_length, sepal_width, petal_length, petal_width]], columns=['sepal_length', 'sepal_width', 'petal_length', 'petal_width'])

# Call the predict_species function with user input data
predictions = predict_species(input_data)
print("Predicted sepal width:", predictions[0])


Enter sepal length: 2.2
Enter sepal width: 2.2
Enter petal length: 2.2
Enter petal width: 2.2
Predicted sepal width: 2.2000000000000006


# In the first code snippet (classification), the model predicts the species of the iris flower based on the given input features (sepal length, sepal width, petal length, petal width). The predicted species is "versicolor" in your example. It classifies the input into one of the three species: "setosa", "versicolor", or "virginica".

# In the second code snippet (regression), the model predicts the sepal width of the iris flower based on the given input features. The predicted sepal width is a numerical value, which in your example is 2.2000000000000006. It treats the problem as a regression task, aiming to predict a continuous value.

![Machine Learning Pipeline](images/1.png)
